### PR TITLE
WAITP-1211: Dashboard Navigation

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Add a redirect for routes that don't have a dashboard name included
- Handle when the dashboard name from the url is not in the dashboards data. This can happen for permission reasons or an incorrect url

### CODE REVIEW

File: packages/tupaia-web-server/src/routes/DashboardsRoute.ts [hunk 1]
```diff
@@ -17,9 +17,10 @@ export class DashboardsRoute extends Route<DashboardsRequest> {
     const project = (
       await ctx.services.central.fetchResources('projects', {
         filter: { code: projectCode },
-        columns: JSON.stringify(['entity.code', 'entity_hierarchy.name']),
```
Comment (tcaiger) [inline packages/tupaia-web-server/src/routes/DashboardsRoute.ts:20]: this is temporary code that Ethan is refactoring in another PR so I wouldn't be too worried about it

File: packages/tupaia-web/src/constants/constants.ts [hunk 2]
```diff
@@ -3,9 +3,6 @@
  * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
  */
 
-export const DEFAULT_PROJECT_CODE = 'explore';
-export const DEFAULT_ENTITY_CODE = 'explore';
-
```
Comment (tcaiger) [inline packages/tupaia-web/src/constants/constants.ts:8]: these consts are only used in the default url so I'm just inlining them below as I think the concept of default project and entity doesn't make sense outside of the default url

File: packages/tupaia-web/src/features/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -99,15 +99,27 @@ const Chart = styled.div`
...
+
+  let activeDashboard = null;
+
+  if (dashboards.length > 0) {
+    activeDashboard =
+      dashboards.find(dashboard => dashboard.name === dashboardName) || dashboards[0];
+  }
+
+  return { dashboards, activeDashboard };
+};
```
Comment (acdunham) [inline packages/tupaia-web/src/features/Dashboard/Dashboard.tsx:114]: Nice, I like this hook

**Review by acdunham — APPROVED** [review]
Nice, this is great. Thanks for doing this!

**Pull request merged**